text/0026-activation-distribution.md

```diff
@@ -0,0 +1,102 @@
+# Project title
+- Start Date: 2020/04/27
+- RFC PR: https://github.com/openmrs/openmrs-rfc-frontend/pull/26
+
+## Decision, including impact on distributions
+For being able to properly distribute the microfrontends the activity function
+has to be distributed, too. Consequently, each microfrontend will be even more
+autonomous and easier to develop. Lastly, it will also be very easy to create a
+distribution using the microfrontend modules.
+
+There are two steps to implement a distribution of the activity functions:
+
+1. Have a proper root module in the microfrontend modules, which defines the
+activity function and lazy loads the rest of the microfrontend (via bundle
+splitting).
+2. Export special functionality that correctly identifies the module as a
+microfrontend that can be properly integrated.
+
+The process to migrate from a centralized module containing all activity
+functions to a distributed starts with a new `index.ts` in the microfrontends.
+
+The `index.ts` may be as simple as follows:
+
+```js
+// set the public path to be able to retrieve side bundles
+import "./set-public-path";
+// register the application in the shared single-spa package
+import { registerApplication } from "single-spa";
+// get useful helpers for the activity functions from the loaded root-config
+import { routePrefix } from '@openmrs/esm-root-config';
+
+// perform the registration with the lazy loading of the lifecycle and its activity function
+registerApplication(
+  "@openmrs/esm-home",
+  () => import('./openmrs-esm-home'),
+  location => routePrefix("home", location),
+);
+
+// export other dependencies to be used, e.g., by the dev tooling
+export { backendDependencies } from "./openmrs-backend-dependencies";
+```
```

Comment:
Also, I just want to make sure that we're on the same page that this no longer applies; that index.ts should satisfy the interface that you call setupOpenMRS and I call the "microfrontend interface," and it should not call registerApplication. Yeah?

Reply:
Good catch! I'll correct it!